pynicotine/gtkgui/search.py

```diff
@@ -1626,23 +1690,15 @@ def on_filter_entry_icon_press(self, entry, *_args):
         else:
             entry.set_text("")
 
-    def on_clear_filters(self, *_args):
+    def on_clear_undo_filters(self, *_args):
 
-        self.clearing_filters = True
-
-        for widget in self.filter_comboboxes.values():
-            widget.get_child().set_text("")
-
-        self.filter_free_slot_button.set_active(False)
+        self.set_filters(self.filters_undo)
 
         if self.filters_button.get_active():
             self.filter_include_combobox.get_child().grab_focus()
```

Comment:
It is not wanted to grab the focus into the text entry when restoring the filters.

Reply:
Well guess so yes, albeit that does not really provide much value either as the current selection is not remembered when refiltering, but it would be less disconcerting.

It

Reply:
Leaving this as-is for now, can be improved in the future if necessary.

Reply:
Sure, maybe now the button does have a function, it would be no harm to allow it to be focused, such that the space bar can be used to toggle clear/restore.